### ISSUE TITLE
Unable to test arbitrary commits

### ISSUE THREAD

**Issue by JustinAzoff** [bro/package-manager#28]
Hi!

I worked out a docker image that contains bro-2.5.3+source+bro-pkg so I can get CI working for packages.  It works locally but unfortunately it seems the way travis tests is by checking out the specific commit first which causes issues:

```
$ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
$ docker run -t -i --rm -v `pwd`:/bro-is-darknet  broplatform/bro:2.5.3-dev bro-pkg --version
bro-pkg 1.3.3
$ docker run -t -i --rm -v `pwd`:/bro-is-darknet  broplatform/bro:2.5.3-dev bro-pkg test /bro-is-darknet
all 7 tests successful
/bro-is-darknet: all tests passed
```

But if an individual commit is checked out, it breaks:

```
$ git checkout 0b45bb4af723f56d7a0b633a889cc64a9bef825b
Note: checking out '0b45bb4af723f56d7a0b633a889cc64a9bef825b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 0b45bb4... enable CI
$ docker run -t -i --rm -v `pwd`:/bro-is-darknet  broplatform/bro:2.5.3-dev bro-pkg test /bro-is-darknet
Traceback (most recent call last):
  File "/usr/local/bin/bro-pkg", line 2007, in <module>
    main()
  File "/usr/local/bin/bro-pkg", line 2003, in main
    args.run_cmd(manager, args, config)
  File "/usr/local/bin/bro-pkg", line 313, in cmd_test
    version = args.version if args.version else active_git_branch(name)
  File "/usr/local/bin/bro-pkg", line 229, in active_git_branch
    rval = repo.active_branch
  File "/usr/local/lib/python2.7/dist-packages/git/repo/base.py", line 690, in active_branch
    return self.head.reference
  File "/usr/local/lib/python2.7/dist-packages/git/refs/symbolic.py", line 275, in _get_reference
    raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
TypeError: HEAD is a detached symbolic reference as it points to '0b45bb4af723f56d7a0b633a889cc64a9bef825b'
```

**Comment by JustinAzoff** [comment]
I was able to workaround it for now by setting the test command to

    ... bro-pkg test /bro-is-darknet --version $(git rev-parse HEAD)

**Comment by jsiwek** [comment]
Thanks, should be fixed by c638122dc694b5efa13233a52122b41baa9c980c and in bro-pkg v1.3.4.
